tools/disassemble_sequence.py

```diff
@@ -0,0 +1,792 @@
+    0xF8: ['loop', 'u8'],
+    0xF7: ['loopend'],
+    0xF6: ['break'],
+    0xF5: ['bgez', 'addr'],
+    0xF4: ['rjump', 'reladdr8'],
+    0xF3: ['rbeqz', 'reladdr8'],
+    0xF2: ['rbltz', 'reladdr8'],
+}
+
+commands = {}
+commands['seq'] = join(control_flow_commands, {
+    # non-arg commands
+    0xF1: ['allocnotelist', 'u8'],
+    0xF0: ['freenotelist'],
+    0xDF: ['transpose', 's8'],
+    0xDE: ['rtranspose', 's8'],
+    0xDD: ['tempo', 'u8'],
+    0xDC: ['tempochg', 's8'],
+    0xDB: ['vol', 'u8'],
+    0xDA: ['volmode', 's8'],
+    0xD7: ['initchan', 'hex16'],
+    0xD6: ['freechan', 'hex16'],
+    0xD5: ['mutescale', 's8'],
+    0xD4: ['mute'],
+    0xD3: ['mutebhv', 'hex8'],
+    0xD2: ['ldshortvelarr', 'addr'],
+    0xD1: ['ldshortgatearr', 'addr'],
+    0xD0: ['notealloc', 'u8'],
+    0xCE: ['rand', 'u8'],
+    0xCD: ['dyncall', 'addr'],
+    0xCC: ['ldi', 'u8'],
+    0xC9: ['and', 'u8'],
+    0xC8: ['sub', 'u8'],
+    0xC7: ['stseq', 'u8', 'addr'],
+    0xC6: ['stop'],
+    0xC5: ['scriptctr', 'u16'],
+    0xC4: ['runseq', 'u8', 'u8'],
+    # arg commands
+    0x00: ['testchan', 'bits:4'],
+    0x40: ['stopchan', 'bits:4'],
+    0x50: ['subio', 'bits:4'],
+    0x60: ['ldres', 'bits:4', 'u8', 'u8'],
+    0x70: ['stio', 'bits:4'],
+    0x80: ['ldio', 'bits:4'],
+    0x90: ['ldchan', 'bits:4', 'addr'],
+    0xA0: ['rldchan', 'bits:4', 'reladdr16'],
+    0xB0: ['ldseq', 'bits:4', 'u8', 'addr'],
+})
+
+commands['chan'] = join(control_flow_commands, {
+    # non-arg commands
+    0xF1: ['allocnotelist', 'u8'],
+    0xF0: ['freenotelist'],
+    0xEE: ['bendfine', 's8'],
+    0xED: ['gain', 'u8'],
+    0xEB: ['fontinstr'],
+    0xEA: ['stop'],
+    0xE9: ['notepri', 'u8'],
+    0xE8: ['params', 'u8', 'u8', 'u8', 's8', 's8', 'u8', 'u8', 'u8'],
+    0xE7: ['ldparams', 'addr'],
+    0xE6: ['samplbook', 'addr'],
+    0xE5: ['rvrbidx', 'u8'],
+    0xE4: ['dyncall'],
+    0xE3: ['vibdelay', 'u8'],
+    0xE2: ['vibdepthgrad', 'u8', 'u8', 'u8'],
+    0xE1: ['vibfreqgrad', 'u8', 'u8', 'u8'],
+    0xE0: ['volexp', 'u8'],
+    0xDF: ['vol', 'u8'],
+    0xDE: ['freqscale', 'u16'],
+    0xDD: ['pan', 'u8'],
+    0xDC: ['panweight', 'u8'],
+    0xDB: ['transpose', 's8'],
+    0xDA: ['env', 'addr'],
+    0xD9: ['releaserate', 'u8'],
+    0xD8: ['vibdepth', 'u8'],
+    0xD7: ['vibfreq', 'u8'],
+#    0xD6: ['setupdatesperframe_unimplemented', 'u8'],
+    0xD4: ['reverb', 'u8'],
+    0xD3: ['bend', 's8'],
+    0xD2: ['sustain', 'u8'],
+    0xD1: ['notealloc', 'u8'],
+    0xD0: ['effects', 'u8'],
+    0xCF: ['stptrtoseq', 'addr'],
+    0xCE: ['ldptr', 'hex16'],
+    0xCD: ['stopchan', 'u8'],
+    0xCC: ['ldi', 'u8'],
+    0xCB: ['ldseq', 'addr'],
+    0xCA: ['mutebhv', 'hex8'],
+    0xC9: ['and', 'u8'],
+    0xC8: ['sub', 'u8'],
+    0xC7: ['stseq', 'u8', 'addr'],
+    0xC6: ['font', 'u8'],
+    0xC5: ['dyntbllookup'],
+    0xC4: ['noshort'],
+    0xC3: ['short'],
+    0xC2: ['dyntbl', 'addr'],
+    0xC1: ['instr', 'u8'],
+    0xBD: ['randptr', 'u16', 'u16'],
```

Comment:
making a note that `randptr` is the one instruction that changes OP in MM compared to OoT, any thoughts how to handle this? (doesn't need addressing here)

Reply:
TBD. What does OP mean in this context?